### ISSUE TITLE
Broken x_forwarded_for in access logs

### ISSUE THREAD

**Issue by giner** [cloudfoundry/gorouter#193]
## Issue

If a request contains "Connection: close" then x_forwarded_for is not logged correctly.

## Context

cf-release 254

## Steps to Reproduce

Run the following command for an existent route:
```
curl -H "Host: xfftest.domain.net" -H "X-Forwarded-For: 1.1.1.1" -H "X-Forwarded-For: 2.2.2.2" -H "Connection: close" localhost:8081
```
And check application logs.

## Expected result

x_forwarded_for should have all IPs joined with commas
```
2017-10-02T19:16:18.78+0900 [RTR/0]      OUT xfftest.domain.net - [2017-10-02T10:16:18.778+0000] "GET / HTTP/1.1" 200 0 61 "-" "curl/7.35.0" "127.0.0.1:16372" "100.73.63.129:61419" x_forwarded_for:"1.1.1.1, 2.2.2.2, 127.0.0.1" x_forwarded_proto:"http" vcap_request_id:"87f77716-b0c6-4597-5c8e-d54f58bf7e00" response_time:0.002689487 app_id:"283c0bf9-1fdd-4b48-94b4-be6bf772c76f" app_index:"0"
```

## Current result

x_forwarded_proto contains only IPs from the first header X-Forwarded-For
```
2017-10-02T19:16:43.16+0900 [RTR/0]      OUT xfftest.domain.net - [2017-10-02T10:16:43.160+0000] "GET / HTTP/1.1" 200 0 61 "-" "curl/7.35.0" "127.0.0.1:17014" "100.73.63.129:61419" x_forwarded_for:"1.1.1.1" x_forwarded_proto:"http" vcap_request_id:"d17061e4-3922-4846-6dc7-ae02e6e2fc10" response_time:0.003125079 app_id:"283c0bf9-1fdd-4b48-94b4-be6bf772c76f" app_index:"0"
```

**Comment by CAFxX** [comment]
> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.

http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

**Comment by shalako** [comment]
Prioritized for eng review. @giner are you able to send us a PR?

**Comment by giner** [comment]
Hi @shalako, if I'm getting it right the issue is here https://github.com/golang/go/commit/3c0f69a52101c69b5a8288195fa74c7ecfa2fa43. X-Forwarded-For brakes if the initial request contains any of [hop by  hop headers](https://github.com/golang/go/blob/3c0f69a52101c69b5a8288195fa74c7ecfa2fa43/src/net/http/httputil/reverseproxy.go#L129-L139).

**Comment by charleshansen** [comment]
We have started upgrading gorouter to 1.9. The issue does appear to be fixed.

**Comment by shalako** [comment]
After upgrading to golang 1.9, it appears the issue is resolved.

on gorouter
```
$ curl -H "Host: dora-scoen.superman.cf-app.com" -H "X-Forwarded-For: 1.1.1.1" -H "X-Forwarded-For: 2.2.2.2" -H "Connection: close" localhost:80
Hi, I'm Dora
```

from workstation
```
$ cf logs dora-scoen
Retrieving logs for app dora-scoen in org scoen / space test as scoen...

   2017-10-19T21:24:53.88-0700 [RTR/0] OUT dora-scoen.superman.cf-app.com - [2017-10-20T04:24:53.875+0000] "GET / HTTP/1.1" 200 0 13 "-" "curl/7.35.0" "127.0.0.1:43174" "10.0.32.14:61002" x_forwarded_for:"1.1.1.1, 2.2.2.2, 127.0.0.1" x_forwarded_proto:"http" vcap_request_id:"cd7edee6-886b-4290-51ba-cd03035ecdaa" response_time:0.005739814 app_id:"27a49ecb-b63c-4315-9159-da7b69ba9016" app_index:"0" x_b3_traceid:"2a50038a77c7c5ae" x_b3_spanid:"2a50038a77c7c5ae" x_b3_parentspanid:"-"
   2017-10-19T21:24:53.88-0700 [RTR/0] OUT
```

**Comment by shalako** [comment]
Fixed in 0.165.0
